test/test_tensorflow.py

```diff
@@ -383,6 +383,114 @@ def test_horovod_allgather(self):
                             tf.equal(tf.cast(rank_tensor, tf.int32), value))),
                         "hvd.allgather produces incorrect gathered tensor")
 
+    def test_horovod_allgather_fused(self):
+        """Test that the allgather correctly gathers 1D, 2D, 3D tensors
+        with Tensor Fusion."""
+        hvd.init()
+        rank = hvd.rank()
+        size = hvd.size()
+
+        with self.test_session(config=self.config) as session:
+            dtypes = [tf.uint8, tf.int8, tf.uint16, tf.int16,
+                      tf.int32, tf.int64, tf.float16, tf.float32,
+                      tf.float64, tf.bool]
+            dims = [1, 2, 3]
+            tests = []
+            shape_tests = []
+            for dtype, dim in itertools.product(dtypes, dims):
+                tensor = tf.ones([17] * dim) * rank
+                if dtype == tf.bool:
+                    tensor = tensor % 2
+                tensor = tf.cast(tensor, dtype=dtype)
+                gathered = hvd.allgather(tensor)
+
+                shape_tests.append(
+                    tf.reduce_all(tf.equal(tf.shape(gathered),
+                                           [17 * size] + [17] * (dim - 1))))
+
+                for i in range(size):
+                    rank_tensor = tf.slice(gathered,
+                                           [i * 17] + [0] * (dim - 1),
+                                           [17] + [-1] * (dim - 1))
+                    # self.assertEqual(list(rank_tensor.shape), [17] * dim)
```

Comment:
Remove self.assertEqual() comment, move tf.equal comment closer to where you do the cast?